src/objective-c/GRPCClient/private/GRPCHost.m

```diff
@@ -127,9 +205,14 @@ - (GRPCChannel *)channel {
         args[@GRPC_SSL_TARGET_NAME_OVERRIDE_ARG] = _hostNameOverride;
       }
 
-      _channel = [GRPCChannel secureChannelWithHost:_address
-                                 pathToCertificates:_pathToCertificates
-                                        channelArgs:args];
+      @synchronized(self) {
+        if (_channelCreds == nil) {
+          [self setSSLCredentials:nil withPrivateKey:nil withCertChain:nil error:nil];
+        }
+        _channel = [GRPCChannel secureChannelWithHost:_address
+                                          credentials:_channelCreds
+                                          channelArgs:args];
+      }
```

Comment:
It's preferable to leave calls to other classes out of the `synchronized` block, so that a reader can easily tell there's no possibility of deadlock. Something like:
```objc
grpc_channel_credentials *channelCreds = NULL;
@synchronized(self) {
  if (!_channelCreds) {
    // Use default root certificates.
    [self setSSLCredentials:nil withPrivateKey:nil withCertChain:nil error:nil];
  }
  channelCreds = _channelCreds;
}
_channel = [GRPCChannel secureChannelWithHost:_address
                                  credentials:channelCreds
                                  channelArgs:args];
```